cpp/src/aztec/honk/sumcheck/sumcheck.test.cpp

```diff
@@ -69,30 +90,203 @@ TEST(Sumcheck, Prover)
+                                                                     /* num_challenges_in = */ 1));
+    for (size_t i = 0; i < multivariate_d; i++) {
+        auto label = std::to_string(multivariate_d - i);
+        manifest_rounds.emplace_back(transcript::Manifest::RoundManifest({ { .name = "univariate_" + label,
+                                                                             .num_bytes = fr_size * max_relation_length,
+                                                                             .derived_by_verifier = false } },
+                                                                         /* challenge_name = */ "u_" + label,
+                                                                         /* num_challenges_in = */ 1));
+    }
+
+    auto transcript = Transcript(transcript::Manifest(manifest_rounds));
+    auto mock_transcript = [](Transcript& transcript) {
+        static_assert(multivariate_d < 64);
+        uint64_t multivariate_n = 1 << multivariate_d;
+        transcript.add_element("circuit_size",
+                               { static_cast<uint8_t>(multivariate_n >> 24),
+                                 static_cast<uint8_t>(multivariate_n >> 16),
+                                 static_cast<uint8_t>(multivariate_n >> 8),
+                                 static_cast<uint8_t>(multivariate_n) });
+    };
+
+    mock_transcript(transcript);
+    transcript.apply_fiat_shamir("alpha");
+
+    auto multivariates = Multivariates(full_polynomials);
+
+    auto sumcheck_prover = Sumcheck<Multivariates,
+                                    Transcript,
+                                    ArithmeticRelation,
+                                    // GrandProductComputationRelation,
+                                    GrandProductInitializationRelation>(multivariates, transcript);
+
+    sumcheck_prover.execute_prover();
+
+    auto sumcheck_verifier = Sumcheck<Multivariates,
+                                      Transcript,
+                                      ArithmeticRelation,
+                                      //   GrandProductComputationRelation,
+                                      GrandProductInitializationRelation>(transcript);
+
+    bool verified = sumcheck_verifier.execute_verifier();
+    ASSERT_TRUE(verified);
+}
+
+TEST(Sumcheck, ProverAndVerifierLonger)
+{
+    auto run_test = [](bool expect_verified) {
+        const size_t num_polys(proving_system::StandardArithmetization::NUM_POLYNOMIALS);
+        const size_t multivariate_d(2);
+        const size_t multivariate_n(1 << multivariate_d);
+        // const size_t max_relation_length = 5;
+
+        const size_t max_relation_length = 4 /* honk::StandardHonk::MAX_RELATION_LENGTH */;
+        constexpr size_t fr_size = 32;
+
+        using Multivariates = ::Multivariates<FF, num_polys>;
+
+        // clang-format off
+    std::array<FF, multivariate_n> w_l;
```

Comment:
Weird choice of initialization was very useful for debugging one of the composer tests.